### DISCUSSION TITLE
Ask questions

### DISCUSSION THREAD

**Discussion by chros098** [The-OpenROAD-Project/OpenROAD-flow-scripts#1451] [General] [answered]
Hello, I am using openroad to add a new process. I refer to the process of nangate45, but I found that several files are not displayed in the process of adding a process, but there are in the platform path.
May I ask what their functions are?
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/142307210/2c725130-cf37-413d-b377-6e2ab8fab681)

**Comment by maliberty** [comment, marked as answer]
cells_adders.v is for yosys to techmap  full/half adders to particular library cells.  It is optional but will help the QOR if you have such cells.

There are no real RAMs for ng45 so the fakeram.* are to generate dummy placeholders.  If you have a RAM generator for your process you don't need these.

magic.tech is for using magic for DRC checking.  It is optional.

rcx_patterns.rules is for RC extraction and is described in the bring up docs.

The template_pga.cfg is obsolete.

**Comment by chros098** [comment]
thanks
